backend/internal/group/handler/grouphandler.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package handler provides the implementation for group management operations.
+package handler
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/asgardeo/thunder/internal/group/model"
+	"github.com/asgardeo/thunder/internal/group/provider"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+// GroupHandler is the handler for group management operations.
+type GroupHandler struct {
+}
+
+// HandleGroupListRequest handles the list groups request.
+func (gh *GroupHandler) HandleGroupListRequest(w http.ResponseWriter, r *http.Request) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "GroupHandler"))
+
+	groupProvider := provider.NewGroupProvider()
+	groupService := groupProvider.GetGroupService()
+	groups, err := groupService.GetGroupList()
+	if err != nil {
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
```

Comment:
Let's add logs for these kind of scenarios. Otherwise we won't be able to find the causes.
Check other places as well